driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```diff
@@ -0,0 +1,176 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.datastax.driver.core;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import com.datastax.driver.core.exceptions.PagingStateException;
+import com.datastax.driver.core.utils.Bytes;
+
+/**
+ * The paging state of a query.
+ *
+ * This object represents the next page to be fetched if the query is
+ * multi page. It can be saved and reused later on the same statement.
+ *
+ * The PagingState can be serialized and deserialized either in a String
+ * or in a ByteBuffer.
+ */
+public class PagingState {
+
+    private ByteBuffer pagingState;
```

Comment:
Instead of keeping the pagingState and hash together and then extracting them separately in different methods, couldn't they be maintained separately as byte arrays (since the values don't really change / they are really small) and then generate the combined buffer whenever toString() and toBytes() are called?

It seems more likely/as likely that getRawState() would be called over toString()/toBytes() so it seems like there would be more value in keeping the two (pagingState and hash) in separate structures and then only combining them as needed.  This would also eliminate the need to rewind the buffers in different methods.